src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List getAll(Class type) {
+        MachineProducer machineProducer;
+        if (type.equals(Bulldozer.class)) {
+            machineProducer = new BulldozerProducer();
+            return machineProducer.get();
+        }
+        if (type.equals(Truck.class)) {
+            machineProducer = new TruckProducer();
+            return machineProducer.get();
+        }
+        if (type.equals(Excavator.class)) {
+            machineProducer = new ExcavatorProducer();
+            return machineProducer.get();
+        }
+        return Collections.emptyList();
```

Comment:
(Optional)
```suggestion
        MachineProducer machineProducer;
        if (type.equals(Bulldozer.class)) {
            machineProducer = new BulldozerProducer();
        } else if (type.equals(Truck.class)) {
            machineProducer = new TruckProducer();
        } else if (type.equals(Excavator.class)) {
            machineProducer = new ExcavatorProducer();
        }
        return machineProducer ? machineProducer.get() : Collections.emptyList();
```

Reply:
IDEA highlights an error in the Ternary operator in return:   Required type: boolean for, I don't know how to check if type equals some class of MachineProducer

Reply:
Sorry, a bad habit from typescript:smiling_face_with_tear:

Reply:
there should be `machineProducer != null `instead of `machineProducer`

Reply:
IDEA said that I have to initialize the machineProduser. So I initialized it with null. Tests completed successfully.